modules/BrainCRISPR.py

```diff
@@ -0,0 +1,278 @@
+
+        Args:
+          study_id (str): The study_id parameter is a string that represents the unique identifier for a
+        single screen.
+
+        Returns:
+          a DataFrame with hit genes from a single screen. The DataFrame contains columns
+            - targetFromSourceId,
+            - pValue,
+            - statisticalTestTail,
+            - resourceScore, and
+            - studyId.
+
+        The function filters out non-hit genes and renames some columns using the PySpark functions col and lit.
+        """
+        screen_url = f"https://storage.googleapis.com/crisprbrain.appspot.com/api-data/{quote(study_id)}.csv.gz"
+        self.spark.sparkContext.addFile(screen_url)
+
+        return (
+            self.spark.read.csv(
+                SparkFiles.get(f"{study_id}.csv.gz"), header=True, inferSchema=True
+            )
+            .filter(f.col("Hit Class") != "Non-Hit")
+            .select(
+                f.col("Gene").alias("targetFromSourceId"),
+                f.col("P Value").alias("resourceScore"),
+                f.when(f.col("Hit Class") == "Positive Hit", f.lit("upper tail"))
+                .when(f.col("Hit Class") == "Negative Hit", f.lit("lower tail"))
+                .alias("statisticalTestTail"),
+                f.col("Phenotype").alias("log2FoldChangeValue"),
+                f.lit(study_id).alias("studyId"),
+            )
+        )
+
+    def __get_all_hit_genes(self, study_identifiers: List[str]) -> DataFrame:
+        """Retrieving hit genes from all screens."""
+
+        # Parsing all gene lists for all screens and combine them into a single dataframe:
+        return reduce(
+            lambda df1, df2: df1.unionByName(df2),
+            map(self.__parse_gene_list, study_identifiers),
+        )
+
+    def create_crispr_brain_evidence(self: CRISPRBrain) -> None:
+        """"""
+        # Reading
+        crispr_studies = (
+            # Reading studies:
+            self.__get_study_table()
+            # Joining with resolved literature:
+            .join(self.__get_literature_mapping(), on="Reference Link", how="left")
+            .select(
+                # Adding constants:
+                f.lit("affected_pathway").alias("datatypeId"),
+                f.lit("crispr_screen").alias("datasourceId"),
+                f.lit("crispr_brain").alias("projectId"),
+                # Renaming fields:
+                f.col("Screen Name").alias("studyId"),
+                f.col("Libraries Screened").alias("crisprScreenLibrary"),
+                f.col("studySummary.title").alias("studyOverview"),
+                # Dropping experiment column for now:
+                # f.col("studySummary.experiment").alias("experiment"),
+                f.col("Cell Type").alias("cellType"),
+                f.when(f.col("Genotype") != "WT", f.col("Genotype"))
+                .otherwise(None)
+                .alias("geneticBackground"),
```

Comment:
This is not strictly necessary, but I think it would be better to make this section slightly easier to read, maybe something like this:

```suggestion
                f.when(f.col("Genotype") != "WT", f.col("Genotype"))
                    .otherwise(None)
                    .alias("geneticBackground"),
```